### PR TITLE
Database setup

### PR DESCRIPTION
**Pull request by fasaldana**
# Blog app - creating a data model

- Create a new app. Give it a meaningful name.
- Initialize your project with Git.
- Make sure that your project has Postgres database set up.
- Build your project schema.
- - Create and run the necessary migration files.
- - Table and column names should match the ERD diagram.
- - - note: photo for users table should be a link to an image
- - Foreign keys should be included.
- - All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: db/migrate/20220707162303_create_post.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreatePost < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.string :test
```
Comment (ahmetbozaci) [inline db/migrate/20220707162303_create_post.rb]: - [x] It should be text not test. Kindly change it.
- [x] Also type should be text not string. Please change it.

File: db/migrate/20220707165348_add_user_ref_to_post.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPost < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, foreign_key: true
+  end
+end
```
Comment (ahmetbozaci) [inline db/migrate/20220707165348_add_user_ref_to_post.rb]: - [x] It should be the author, not the user. Kindly change it.

You can do it like in the picture.
image

File: db/migrate/20220707171901_add_user_ref_to_comment.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComment < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, foreign_key: true
```
Comment (ahmetbozaci) [inline db/migrate/20220707171901_add_user_ref_to_comment.rb]: - [x] It should be the author, not the user. Kindly change it.

You can do it like in the picture.
image

File: db/migrate/20220707172003_add_user_ref_to_likes.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, foreign_key: true
+  end
```
Comment (ahmetbozaci) [inline db/migrate/20220707172003_add_user_ref_to_likes.rb]: - [x] It should be the author, not the user. Kindly change it.
[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_1.md#:~:text=Table%20and%20column%20names%20should%20match%20the%20ERD%20diagram.)
You can do it like in the picture.
image

File: db/migrate/20220707162453_create_comment.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class CreateComment < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.string :text
+
```
Comment (ahmetbozaci) [inline db/migrate/20220707162453_create_comment.rb]: - [x] Type should be text not string. Kindly change it

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @fasaldana  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- Your schema is almost finished according to requirements ✔️
___


### ♻️ Request changes ♻️

- [x] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by julie-ify — APPROVED** [review]
Hi @fasaldana, 

- [ ] **[OPTIONAL]** Kindly add a descriptive Readme to your repository, it is very important to have the Readme so please add it by making use of the Microverse Readme [template](https://github.com/microverseinc/readme-template)

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**